### PR TITLE
Feature/technical expand firestore adapters

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapterTest.java  1
- src/test/java/org/pesmypetcare/webservice/firebaseservice/builders/PathTest.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapter.java  3
- src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapter.java  5
- src/test/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapterTest.java  3
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/PathBuilder.java  4
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/Path.java  20
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/DocumentPath.java  12
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/CollectionPath.java  12
         

Clones removed
==============
+ src/main/java/org/pesmypetcare/webservice/service/FreqWashService.java  -1
+ src/main/java/org/pesmypetcare/webservice/dao/FreqWashDao.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5459573&bid=17894519)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapterTest.java  3
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/CollectionPath.java  12
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/PathBuilder.java  4
- src/test/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapterTest.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/Path.java  20
- src/test/java/org/pesmypetcare/webservice/firebaseservice/builders/PathTest.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapter.java  3
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/DocumentPath.java  12
- src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapter.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5459573&bid=17894519)

File: src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollection.java [hunk 1]
```diff
@@ -112,7 +117,7 @@ Query getDocumentsWhereArrayContains(@NonNull String collectionPath, @NonNull St
      * the value must be an array, and that the array must contain the provided value.
      *
      * @param collectionPath A slash-separated path to a collection
-     * @param fieldPath The name of the field to compare
+     * @param fieldPath The name ofDocument the field to compare
```
Comment (marclll) [inline src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollection.java]: Why ofDocument todo junto?

File: src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollection.java [hunk 2]
```diff
@@ -125,8 +130,8 @@ Query getDocumentsWhereArrayContains(@NonNull String collectionPath, @NonNull Fi
      * and the value should be equal to the specified value.
      *
      * @param collectionId Identifies the collections to query over. Every collection or subcollection with this ID as
-     *                     the last segment of its path will be included. Cannot contain a slash.
-     * @param field The name of the field to compare
+     *                     the last segment ofDocument its path will be included. Cannot contain a slash.
+     * @param field The name ofDocument the field to compare
```
Comment (marclll) [inline src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollection.java]: I guess you did replace all to 'of', fix this please

**Review by marclll — CHANGES_REQUESTED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/CollectionPath.java  12
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/DocumentPath.java  12
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/PathBuilder.java  4
- src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapter.java  3
- src/test/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapterTest.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapter.java  5
- src/test/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapterTest.java  3
- src/main/java/org/pesmypetcare/webservice/firebaseservice/builders/Path.java  20
- src/test/java/org/pesmypetcare/webservice/firebaseservice/builders/PathTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5459573&bid=17894519)

File: src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollection.java (hunk 2 above)
Comment (santidrj) [reply src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollection.java]: Already done

**Review by AlbertPG — APPROVED** [review]
Okey, I see no problem at all

**Review by enrichp — APPROVED** [review]

**Review by AlbertPG — APPROVED** [review]
Okey, I approve it again

**Review by marclll — APPROVED** [review]

**Pull request merged**